### DISCUSSION TITLE
[ CRUD service ] - Read

### DISCUSSION THREAD

**Discussion by georgev-97** [mia-platform/community#478] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12

### Description

Good morning guys,

I have many services that write directly on mongo without setting` __STATE__: PUBLIC`
the CRUD won't find those entries.
Is there a way to retrieve this entries via the CRUD?

Forcing the variable `__STATE__ `on insert is way too complex

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @georgev-97 

The CRUD service performs operations over documents having the proper [metadata fields](https://github.com/mia-platform/crud-service?tab=readme-ov-file#metadata-fields) (`__STATE__` is the one only mandatory).

If it's possible, you should make your services insert your records through the [create endpoints provided by CRUD itself](https://docs.mia-platform.eu/docs/runtime_suite/crud-service/overview_and_usage#create), to support the proper metadata.

If you can't edit the services to switch from MongoDB driver to crud service HTTP calls, you can expose these collections by using the [MongoDB Views definition provided by the console](https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/mongo_views).

Let's suppose you have a collection named `test-coll` with a field named `test`, that is populated by external services.

<img width="1132" alt="image" src="https://github.com/mia-platform/community/assets/38134891/cfcc170c-0cff-4ab6-bd8b-b9549a7febdb">

Then, you can create a MongoDB view named `test-view` using the same fields (using the import/export feature, it should be easy to have the same data model)

<img width="1132" alt="image" src="https://github.com/mia-platform/community/assets/38134891/6173bdd2-9e16-4c39-b5b7-ba440b9ca834">

Finally, in the aggregation section, you can choose `test-coll` and using a `$project` stage to insert a mock `__STATE__` field.

<img width="1132" alt="image" src="https://github.com/mia-platform/community/assets/38134891/63f91c5b-f34c-49ec-bfba-467726bb83a3">

So, you can use the endpoint provided by the view to expose the view rather than the collection, to have a document that is compliant to CRUD Service definitions.

> :warning: **Keep in mind that MongoDb view endpoints expose only `GET` endpoints**. If you need to perform updates using CRUD service endpoints, consider using CRUD service to perform all operations over your collections, or update your services to provide the `__STATE__` field on insert.  

Let me know if you need further assistance on this 😃 

Bests
Giovanni

**Reply by georgev-97** [reply]
Thank you for your response
I think this solution might work for me

Best regards,
George
